OneSignalSDK/onesignal/core/src/main/java/com/onesignal/core/internal/permissions/AlertDialogPrepromptForAndroidSettings.kt

```diff
@@ -53,18 +55,26 @@ object AlertDialogPrepromptForAndroidSettings {
         val messageTemplate = activity.getString(R.string.permission_not_available_message)
         val message = messageTemplate.format(previouslyDeniedPostfix)
 
-        AlertDialog.Builder(activity)
-            .setTitle(title)
-            .setMessage(message)
-            .setPositiveButton(R.string.permission_not_available_open_settings_option) { dialog, which ->
-                callback.onAccept()
-            }
-            .setNegativeButton(android.R.string.no) { dialog, which ->
-                callback.onDecline()
-            }
-            .setOnCancelListener {
-                callback.onDecline()
-            }
-            .show()
+        // ensure the activity that will be showing the dialog is available
+        if (activity == null || activity.isDestroyed || activity.isFinishing) {
+            Logging.log(LogLevel.ERROR, "Alert dialog for Android settings was skipped because the activity was unavailable to display it.")
+            return
```

Comment:
This will prevent the crash but it will create some side effects when this case is hit:
1. Dropped permission requests - Permission prompt won't show when the app is resumed
2. Callback never fires - Customers code may hang for ever thinking the the prompt is still showing.

Here some quick thoughts on soe options to address these:
1. Wait - We can suspend the code here, until an activity is available.
2. Fire onDecline - This is a bit missleading but does solve the problem 2 note above.
* We could add a 3rd callback, something like onSkiped(), but I think that will make things more complex for customer than needed, so really don't like this alt B option here.

Reply:
Good point, I have added a code to fire the onDecline callback because it is consistent with the behavior when the dialog is cancelled or dismissed without a response.